client/scripts/login.ts

```diff
@@ -3,7 +3,30 @@ import * as $ from "jquery"
 namespace loginpage {
 
     function do_login() {
-        return true; //do form submit
+        let email : string = (<HTMLInputElement>document.getElementById("email")).value;
+        let password : string = (<HTMLInputElement>document.getElementById("password")).value;
+
+        /**
+         * There is an official standard for valid email addresses known as RFC 2822
+         * The regex below validates email addresses on this format
+         * If interested, read more here:
+         * https://www.ietf.org/rfc/rfc2822.txt
+         * http://www.regular-expressions.info/email.html
+         **/
+        let regex : string = "[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?";
+
+        if (!email.match(regex)) {
+            alert("Invalid email address.");
+        }
+
+        else if (password.length < 5) {
+            alert("Wrong password.");
+        }
+
+        else {
+            location.href = "main.html";
```

Comment:
I'm pretty sure you don't want to do location.href here, the form will automatically goto the action url (/action/login) when a button inside the form is pressed